internal/git/gitsync_processor.go

```diff
@@ -462,6 +496,48 @@ func NewGitSyncProcessor(ctx context.Context, gitSync *v1alpha1.GitSync, kubeCli
 	return processor, nil
 }
 
+func GetAuthMethod(ctx context.Context, repoUrl string, kubeClient kubernetes.Client, namespace string, repoCred *controllerconfig.GitCredential) (transport.AuthMethod, error) {
+	scheme, err := validations.GetTransportScheme(repoUrl)
+	if err != nil {
+		return nil, err
+	}
+	var authMethod transport.AuthMethod
+	switch scheme {
+	case "ssh", "git+ssh":
```

Comment:
I didn't know about "git+ssh" - do you know if this is the "git" protocol mentioned [here](https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols)? (Interestingly, that link there says there's no authentication for it so it's possible we don't go through this logic for that.) Have you found this used anywhere? Is there some documentation you used in the go-git reference that you can point me to for this?

Reply:
I found this repository -https://github.com/whilp/git-urls/blob/master/urls.go ,it has mentioned git+ssh transports

Reply:
thanks for noticing that